### PR TITLE
updating vnv and ipam cmdlets

### PR DESCRIPTION
**Pull request by sssharma24**
## Mandatory Checklist

- Please choose the target release of Azure PowerShell. (⚠️**Target release** is a different concept from **API readiness**. Please click below links for details.)
  - [x] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] No need for a release

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Review by VeryEarly — COMMENTED** [review]
this looks like breaking change to make an optional parameter mandatory. updating the parameter set name won't help

**Comment by VeryEarly** [comment]
please update changelog: https://github.com/sssharma24/azure-powershell/blob/sssharma/ps-bugbash/src/Network/Network/ChangeLog.md

**Comment by sssharma24** [comment]
> this looks like breaking change to make an optional parameter mandatory. updating the parameter set name won't help

which file/cmdlet are you referring to

**Comment by VeryEarly** [comment]
> > this looks like breaking change to make an optional parameter mandatory. updating the parameter set name won't help
> 
> which file/cmdlet are you referring to

https://github.com/Azure/azure-powershell/blob/85820fce1f3343de4d51780383f786ada9ce1c47/src/Network/Network/NetworkManager/VnetVerifier/GetAzNetworkManagerVerifierWorkspaceReachabilityAnalysisIntentCommand.cs#L35-L43

**Comment by sssharma24** [comment]
> this looks like breaking change to make an optional parameter mandatory. updating the parameter set name won't help

I have reverted it to optional

**Comment by sssharma24** [comment]
> please update changelog: https://github.com/sssharma24/azure-powershell/blob/sssharma/ps-bugbash/src/Network/Network/ChangeLog.md

changelog is already updated all of these cmdlets are new and will be added in this release

**Comment by sssharma24** [comment]
@VeryEarly could you run the pipelines

File: src/Network/Network/NetworkManager/IpamPool/NewAzNetworkManagerIpamPoolStaticCidrCommand.cs [hunk 1]
```diff
@@ -28,57 +28,66 @@
...
-         Mandatory = true,
-         ValueFromPipelineByPropertyName = true,
-         HelpMessage = "IP Address Manager Pool resource name.")]
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "IP Address Manager Pool resource name.",
+            ParameterSetName = CreateByNameParameterSet)]
          [ValidateNotNullOrEmpty]
-        public virtual string IpamPoolName { get; set; }
+        public virtual string PoolName { get; set; }
```
Comment (VeryEarly) [inline src/Network/Network/NetworkManager/IpamPool/NewAzNetworkManagerIpamPoolStaticCidrCommand.cs:70]: changing parameter name is breaking change

File: src/Network/Network/NetworkManager/IpamPool/NewAzNetworkManagerIpamPoolStaticCidrCommand.cs [hunk 2]
```diff
@@ -28,57 +28,66 @@
...
             Mandatory = false,
             ValueFromPipelineByPropertyName = true,
-            HelpMessage = "Number of IP addresses to allocate for a static CIDR resource. The IP addresses will be assigned based on IpamPools available space.")]
+            HelpMessage = "Number of IP addresses to allocate for a static CIDR resource. The IP addresses will be assigned based on IpamPools available space.",
+            ParameterSetName = CreateByNameParameterSet)]
         public virtual string NumberOfIPAddressesToAllocate { get; set; }
 
         [Parameter(
-            Mandatory = false,
+            Mandatory = true,
```
Comment (VeryEarly) [inline src/Network/Network/NetworkManager/IpamPool/NewAzNetworkManagerIpamPoolStaticCidrCommand.cs:80]: this parameter used to be optional with name parameter set, make it mandatory is breaking change

File: src/Network/Network/NetworkManager/IpamPool/NewAzNetworkManagerIpamPoolStaticCidrCommand.cs (hunk 1 above)
Comment (arjun-d-patel) [reply src/Network/Network/NetworkManager/IpamPool/NewAzNetworkManagerIpamPoolStaticCidrCommand.cs:70]: This feature is a new feature and hasn't been released yet. We are correcting the name here from a previous PR: https://github.com/Azure/azure-powershell/pull/26345

**Comment by arjun-d-patel** [comment]
> Commenter does not have sufficient privileges for PR 26654 in repo Azure/azure-powershell

@VeryEarly Looks like we don't have permissions to run the pipeline. Could you help to run this pipeline and also is there any way we can get permissions to run this pipeline?

**Review by VeryEarly — APPROVED** [review]

**Pull request merged**